core/identity/identity.go

```diff
@@ -280,6 +278,32 @@ func (s *service) getIdentitiesDataFromRepo(ctx context.Context, identities []st
 	return res, nil
 }
 
+func (s *service) batchIdentities(ctx context.Context, identities []string) ([]*identityrepoproto.DataWithIdentity, []string) {
+	resultIdentities := make([]*identityrepoproto.DataWithIdentity, 0, len(identities))
+	failedIdentities := make([]string, 0)
+	for j := 0; j < len(identities); {
+		var (
+			repoIdentities      []*identityrepoproto.DataWithIdentity
+			err                 error
+			requestedIdentities []string
+		)
+		if j+identityBatch < len(identities) {
+			requestedIdentities = identities[j : j+identityBatch]
+		} else {
+			requestedIdentities = identities[j:]
+		}
+		repoIdentities, err = s.identityRepoClient.IdentityRepoGet(ctx, requestedIdentities, []string{identityRepoDataKind})
+		if err != nil {
+			log.Error("error getting identity repo data", zap.Error(err))
```

Comment:
I'm concerned about the number of log messages. Is it possible that we will log unnecessary errors?

Reply:
May be we can collect such errors and then log it in one log? 🧐